sycl/test/basic_tests/image_constructors.cpp

```diff
@@ -0,0 +1,344 @@
+// RUN: %clang -std=c++11 %s -o %t1.out -lstdc++ -lOpenCL -lsycl
+// RUN: env SYCL_DEVICE_TYPE=HOST %t1.out
+// RUN: %clang -std=c++11 -fsycl %s -o %t2.out -lstdc++ -lOpenCL -lsycl
+// RUN: env SYCL_DEVICE_TYPE=HOST %t2.out
+// RUN: %CPU_RUN_PLACEHOLDER %t2.out
+// RUN: %GPU_RUN_PLACEHOLDER %t2.out
+// RUN: %ACC_RUN_PLACEHOLDER %t2.out
+//==------------------- image_constructors.cpp - SYCL buffer basic test
+//----------------==//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include <CL/sycl.hpp>
+#include <cassert>
+#include <iostream>
+
+void no_delete(void *) {}
+int main() {
+
+  cl::sycl::image_channel_order channelOrder =
+      cl::sycl::image_channel_order::rgbx;
+  cl::sycl::image_channel_type channelType =
+      cl::sycl::image_channel_type::signed_int32;
+  unsigned int elementSize = 12; // rgbx * i32
+  cl::sycl::range<3> r(3, 3, 3);
+  int numElems = 27;     // 3*3*3
+  int imageHostPtr[144]; // 16*9
+  for (int i = 0; i < 144; i++)
+    imageHostPtr[i] = i;
+  cl::sycl::property_list propList{}; // empty property list
```

Comment:
This test is Ok for now, but in this patch or in a future patch I suggest improving it by moving all that code from main into a template function:

template <int D>
void testConstructors(range<D> r, void *ImageHostPtr) {
  // most of that code form main() here
}
int main() {
  ...
  testConstructors<1>(range{3}, HostData);
  testConstructors<2>(range{3,3}, HostData);
  testConstructors<3>(range{3,3,3}, HostData);
}

Reply:
added the change.

Reply:
More generally I advise not using the same values, because it might hide some transposition bugs, or if you mess up with a combination of x, y, z... For example how to differentiate between 3, 3 and 3? :-)
```
testConstructors<2>(range{3,4}, HostData);
testConstructors<3>(range{3,2,5}, HostData);
```

Reply:
I think you are correct. I did the needed change and have uploaded a new review.